src/analytics/index.ts

```diff
@@ -1,7 +1,132 @@
-import { createClient } from '@segment/analytics-react-native';
-// @ts-ignore
-import { REACT_APP_SEGMENT_API_WRITE_KEY } from 'react-native-dotenv';
+import { createClient, SegmentClient } from '@segment/analytics-react-native';
+import {
+  REACT_APP_SEGMENT_API_WRITE_KEY,
+  LOG_LEVEL,
+  LOG_DEBUG,
+} from 'react-native-dotenv';
 
+import { EventProperties, event } from '@/analytics/event';
+import { UserProperties } from '@/analytics/userProperties';
+import { LogLevel, logger } from '@/logger';
+
+export class Analytics {
+  client: SegmentClient;
+  currentWalletAddressHash?: string;
+  deviceId?: string;
+  event = event;
+  disabled = false;
+
+  constructor() {
+    this.client = createClient({
+      debug: Boolean(LOG_DEBUG) || LOG_LEVEL === LogLevel.Debug,
```

Comment:
Debug mode is controlled via the same mechanism that handles the logger debug mode.